pytorch_lightning/trainer/evaluation_loop.py

```diff
@@ -60,8 +60,14 @@ def get_evaluation_dataloaders(self, max_batches):
         else:
             # val
             in_sanity_check = self.trainer.running_sanity_check
-            should_reload_every_epoch = self.trainer.reload_dataloaders_every_epoch
-            if (self.trainer.val_dataloaders is None or should_reload_every_epoch) and not in_sanity_check:
+
+            reload_dl_every_n_epoch = self.trainer.reload_dataloaders_every_n_epoch
+            if reload_dl_every_n_epoch and (not self.trainer.current_epoch % reload_dl_every_n_epoch):
+                should_reload_every_n_epoch = True
+            else:
+                should_reload_every_n_epoch = False
```

Comment:
Can should_reload_every_n_epoch become a property of the trainer instead ?

Reply:
Yes. I can perform this logic check only in the training_loop, `on_train_epoch_start`, and update a `self.should_reload_every_n_epoch`. Then can I just condition on this variable in evaluation_loop, `get_evaluation_dataloaders`. Would this work?